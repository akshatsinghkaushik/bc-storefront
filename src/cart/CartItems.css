.cart-items-grid {
  width: 100%;
  display: grid;
  grid-template-columns: repeat(1, auto);
  grid-gap: 0;
  margin: 1rem 0 1rem 0;
}

.cart-grid-item {
  grid-row: span 1;
  grid-column: span 1;
  align-self: center;
  justify-self: center;
  display: flex;
  flex-direction: row;
  align-items: center;
  justify-content: space-between;
  width: 45rem;
  height: 8rem;
  margin-bottom: 2rem;
}

.cart-item-image {
  width: 7rem;
  height: 4rem;
  object-fit: cover;
  margin-right: 1rem;
}

.cart-item-text {
  text-align: left;
  text-decoration: none;
  color: black;
}

.cart-item-text h3 {
  letter-spacing: 1px;
}

.cart-item-text p {
  color: gray;
}

.product-color {
  color: black;
}

.product-items {
  display: flex;
  flex-direction: row;
  align-items: center;
  justify-content: space-between;
}

.incre-decre-buttons {
  display: flex;
  flex-direction: column;
  margin: 0 0 0 0.3rem;
}

.incre-button {
  margin: 0 0 0.3rem 0;
}

.total,
.action {
  margin: 0 1rem 0 1rem;
  display: flex;
  align-items: center;
}

.total {
  color: rgb(131, 129, 129);
  font-weight: 600;
}

.total-amount {
  background-color: rgba(223, 223, 223, 0.795);
  padding: 1rem;
  min-width: 1.5rem;
  text-align: center;
}

.cart-details .quantity {
  padding-right: 1rem;
}

.amount .action {
  background: none;
  color: inherit;
  border: none;
  padding: 1rem;
  font: inherit;
  cursor: pointer;
  outline: inherit;
}
