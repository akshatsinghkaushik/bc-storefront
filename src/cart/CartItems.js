import React, { useContext } from 'react';
import { Link } from 'react-router-dom';
import CartContext from '../context/CartContext';
import './CartItems.css';

const CartItems = () => {
  const cartContext = useContext(CartContext);

  const containsColor = (text) => {
    const colors = [
      'Blue',
      'Green',
      'Indigo',
      'Violet',
      'Yellow',
      'Orange',
      'Red',
    ];

    return colors.filter((color) => text && text.includes(color));
  };
  return (
    <div className="cart-items-grid">
      {cartContext.cartMapArr.length === 0 ? (
        <p style={{ textAlign: 'center', padding: '2rem' }}>No Items in Cart</p>
      ) : (
        ''
      )}
      {cartContext.cartMapArr &&
        cartContext.cartMapArr.map((item, index) => (
          <div key={`cart-item-${index}`}>
            <hr style={{ height: '1px', width: '100%' }} />
            <div className="cart-grid-item">
              <div className="product-items">
                <img
                  className="cart-item-image"
                  src={`/media/${item.image}`}
                  alt={item.title}
                />
                <Link
                  to={`/product/${item.title}`}
                  style={{ textDecoration: 'none' }}
                >
                  <div className="cart-item-text">
                    <p>{item.brand}</p>
                    <h3>{item.title}</h3>
                    {containsColor(item.title).length !== 0 ? (
                      <p>
                        Color:{' '}
                        <span className="product-color">
                          {containsColor(item.title)}
                        </span>
                      </p>
                    ) : (
                      ''
                    )}
                  </div>
                </Link>
              </div>
              <div className="amount">
                <div className="quantity">
                  <div className="total-amount">{item.quantity}</div>
                  <div className="incre-decre-buttons">
                    <button
                      className="incre-button"
                      onClick={() =>
                        cartContext.updateCartItemQuantity(item, '+')
                      }
                    >
                      +
                    </button>
                    <button
                      onClick={() =>
                        cartContext.updateCartItemQuantity(item, '-')
                      }
                    >
                      -
                    </button>
                  </div>
                </div>
                <div className="total">
                  $ {Number(item.quantity * item.price).toFixed(2)}
                </div>
                <button
                  className="action"
                  onClick={() => cartContext.deleteCartItem(item)}
                >
                  X
                </button>
              </div>
            </div>
          </div>
        ))}
    </div>
  );
};

export default CartItems;
