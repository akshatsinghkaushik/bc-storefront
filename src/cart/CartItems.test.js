import React from 'react';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import CartItems from './CartItems';
import MockCartProvider from '../tests/MockCartProvider';

it('CartItems Renders correctly', () => {
  const tree = renderer
    .create(
      <BrowserRouter>
        <MockCartProvider>
          <CartItems />
        </MockCartProvider>
      </BrowserRouter>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
