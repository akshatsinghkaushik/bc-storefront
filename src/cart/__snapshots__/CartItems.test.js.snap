// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CartItems Renders correctly 1`] = `
<div
  className="cart-items-grid"
>
  <p
    style={
      Object {
        "padding": "2rem",
        "textAlign": "center",
      }
    }
  >
    No Items in Cart
  </p>
</div>
`;
